src/chunk.c

```diff
@@ -1410,3 +1410,46 @@ chunk_set_schema(Chunk *chunk, const char *newschema)
 
 	return chunk_update_form(&chunk->fd);
 }
+
+/* Used as a tuple found function */
+static bool
+chunk_rename_schema_name(TupleInfo *ti, void *data)
+{
+	HeapTuple	tuple = heap_copytuple(ti->tuple);
+	FormData_chunk *chunk = (FormData_chunk *) GETSTRUCT(tuple);
+
+	/* Rename schema name */
```

Comment:
Since this part is essentially repeated in dimension, is it useful to add a utility func to cataog.c called like `tuple_rename_schema_name`?

Reply:
I thought about refactoring the repeated parts, but then decided against it since you would have to pass many args in (ie, all of the vars in ScannerCtx), so it wasn't clear it would cut code or make the code more readable, but let me stare at it some more and see if refactoring would be useful.

Reply:
Oh, actually, you're saying something different in this comment. I don't think this part is refactorable at all, since you need to know the type of the type you're casting to (as well as the field you're renaming).

Reply:
Sorry I meant those last 3 lines, not the previous part.

Also re:ScannerCtx - you don't need to pass in all the args for ScannerCtx since most of them are the same between the two functions. Just the table, index, and tuple_found parameters.

Reply:
And a parameter to pick the right Anum (e.g. `Anum_chunk_schema_name_idx_schema_name`)

Reply:
Well, these would only be reused for chunk and dimension, not hypertable, but maybe that's what you intended from the beginning?

Reply:
Yes I meant for just these two. Maybe it doesn't make sense for just the two things (since @erimatnor probably would have suggested it before me), but could be useful in the future if other catalog items need schema renames. It's probably not a blocker so I'll remove my block.

Reply:
Hm, ok, leaving the refactoring out of this PR, but we can always go in and fix up in the future if this becomes more unwieldy.